tasks/build/notice.js

```diff
@@ -0,0 +1,53 @@
+import _ from 'lodash';
+import npm from 'npm';
+import npmLicense from 'license-checker';
+import path from 'path';
+import fs from 'fs';
+import { execSync } from 'child_process';
+
+export default function licenses(grunt) {
+  grunt.registerTask('_build:notice', 'Adds a notice', async function () {
+    const overrides = grunt.config.get('licenses.options.overrides');
+    const done = this.async();
+    const buildPath = path.join(grunt.config.get('buildDir'), 'kibana');
+    const noticePath = path.join(buildPath, 'NOTICE.txt');
+    const fd = fs.openSync(noticePath, 'w');
```

Comment:
this should probably be started from a base file, where we can include committed dependencies too